code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -100,19 +94,20 @@
 	return ..()
 
 /obj/item/gun/ballistic/rifle/boltaction/attackby(obj/item/item, mob/user, params)
-	. = ..()
-	if(!can_jam)
-		balloon_alert(user, "can't jam!")
-		return
-
-	if(!bolt_locked)
+	if(!bolt_locked && !istype(item, /obj/item/knife))
```

Comment:
Why are we checking for knife?

Reply:
bayonet attachment reasons so you don't have to have the bolt open to affix a knife to the front of your gun

if it wasn't checked for here then you'd have to have the bolt open to attach a bayonet which doesn't seem right to me